component-catalog-connectors/artifactory-connector/artifactory_catalog_connector/artifactory_catalog_connector.py

```diff
@@ -0,0 +1,293 @@
+
+        try:
+            self.log.debug("Retrieving component list from Artifactory catalog.")
+
+            component_list += recursively_get_components(
+                api_base_url=api_base_url,
+                api_auth=api_auth,
+                repository_name=repository_name,
+                file_filter=file_filter,
+                file_ordering=file_ordering,
+                max_recursion_depth=max_recursion_depth,
+                max_files_per_folder=max_files_per_folder,
+                current_folder_path=repository_path,
+                current_recursion_depth=0,
+            )
+
+        except Exception as ex:
+            self.log.error(
+                f"Error retrieving component list from Artifactory catalog: {ex}"
+            )
+
+        return component_list
+
+    def read_catalog_entry(
+        self, catalog_entry_data: Dict[str, Any], catalog_metadata: Dict[str, Any]
+    ) -> Optional[str]:
+        """
+        Fetch the component that is identified by catalog_entry_data from
+        the Artifactory catalog.
+
+        :param catalog_entry_data: a dictionary that contains the information needed to read the content
+                                   of the component definition
+        :param catalog_metadata: the metadata associated with the catalog in which this catalog entry is
+                                 stored; in addition to catalog_entry_data, catalog_metadata may also be
+                                 needed to read the component definition for certain types of catalogs
+
+        :returns: the content of the given catalog entry's definition in string form
+        """
+        artifactory_username = catalog_metadata.get("artifactory_username")
+        artifactory_password = catalog_metadata.get("artifactory_password")
+
+        url = catalog_entry_data.get("url")
+        if url is None:
+            self.log.error("Artifactory component source must contain `url`.")
+            return None
+
+        api_auth = None
+        if artifactory_username and artifactory_password:
+            api_auth = HTTPBasicAuth(
+                username=artifactory_username, password=artifactory_password
+            )
+        elif artifactory_password:
+            api_auth = ArtifactoryApiKeyAuth(api_key=artifactory_password)
+
+        resp = requests.get(url, auth=api_auth)
+        if resp.status_code != HTTPStatus.OK:
+            self.log.error(
+                f"Failed to read component from '{url}'... "
+                f"Got unhandled HTTP Status {resp.status_code}, expected 200"
+            )
+            return None
+        if resp.headers["Content-Type"] != "text/plain":
+            self.log.error(
+                f"Failed to read component from '{url}'... "
+                f"Got unexpected Content-Type '{resp.headers['Content-Type']}', expected 'text/plain'"
+            )
+            return None
+
+        return resp.text
```

Comment:
To accomodate the new return type for [Kubeflow Pipelines connectors](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/catalog_connector.py#L67):

```suggestion
        return KfpEntryData(definition=resp.text)
```